src/i_video.c

```diff
@@ -1349,6 +1351,49 @@ static void I_InitGraphicsMode(void)
     I_ResetGraphicsMode();
 }
 
+static void I_ReinitGraphicsMode(void)
+{
+    if (texture_upscaled != NULL)
+    {
+        SDL_DestroyTexture(texture_upscaled);
+        texture_upscaled = NULL;
+    }
+
+    if (texture != NULL)
+    {
+        SDL_DestroyTexture(texture);
+        texture = NULL;
+    }
+
+    if (argbbuffer != NULL)
+    {
+        SDL_FreeSurface(argbbuffer);
+        argbbuffer = NULL;
+    }
+
+    if (sdlscreen != NULL)
+    {
+        SDL_FreeSurface(sdlscreen);
+        sdlscreen = NULL;
+    }
```

Comment:
I think we only need to destroy `renderer` and `screen`, everything else is destroyed in `I_ResetGraphicsMode()` anyway:
```diff
diff --git a/src/i_video.c b/src/i_video.c
index f5fa979..f6c3a7b 100644
--- a/src/i_video.c
+++ b/src/i_video.c
@@ -1353,30 +1353,6 @@ static void I_InitGraphicsMode(void)
 
 static void I_ReinitGraphicsMode(void)
 {
-    if (texture_upscaled != NULL)
-    {
-        SDL_DestroyTexture(texture_upscaled);
-        texture_upscaled = NULL;
-    }
-
-    if (texture != NULL)
-    {
-        SDL_DestroyTexture(texture);
-        texture = NULL;
-    }
-
-    if (argbbuffer != NULL)
-    {
-        SDL_FreeSurface(argbbuffer);
-        argbbuffer = NULL;
-    }
-
-    if (sdlscreen != NULL)
-    {
-        SDL_FreeSurface(sdlscreen);
-        sdlscreen = NULL;
-    }
-
     if (renderer != NULL)
     {
         SDL_DestroyRenderer(renderer);
```

Reply:
Fixed by https://github.com/fabiangreffrath/woof/pull/1089/commits/aa894ca58fb1cc7533a7d5774a8963850ade41f3